### ISSUE TITLE
it seems that there is no L2-norm constraints of the entities embedding?

### ISSUE THREAD

**Issue by ocsponge** [thunlp/OpenKE#52]
it seems that there is no L2-norm constraints of the entities embedding in the transE code implementation, while the constraints is recommended in the transE paper, for what consideration you made this change?

**Comment by doug919** [comment]
Same question. Also, the scoring function used in the paper is a square of L2-norm, aka squared sum, but you are using L1-norm for scoring in both TensorFlow and PyTorch implementations. Can you provide some justification for this?

**Comment by Kalethars** [comment]
I'm confused, why L1-norm in TransE?

**Comment by ShulinCao** [comment]
You can use the interface **set_p_norm** in PyTorch branch. 
```
def set_p_norm(self, p_norm):
        self.p_norm = p_norm
```
For L1-norm, you can set p_norm as 1, and 2 for L2-norm.

**Comment by UditGupta10** [comment]
@ShulinCao The original question is still not answered. There seems to be no L2 norm constraints on the entity embedding while optimisation, but the paper has mentioned it.

**Comment by THUCSTHanxu13** [comment]
I have added  normalization operations for TransX models. And the current implementation can achieve the state-of-the-art results on FB15K237 and WN18RR. The hyper-parameters and script code can be founded in the folder "./examples/".
